src/problems/wfg.cpp

```diff
@@ -133,7 +131,7 @@ std::pair<vector_double, vector_double> wfg::get_bounds() const
 {
     vector_double upper_bounds(m_dim_dvs);
     for (decltype(m_dim_dvs) i = 0u; i < m_dim_dvs; ++i) {
-        upper_bounds[i] = 2.0 * (i + 1);
+        upper_bounds[i] = 2.0 * (static_cast<double>(i) + 1);
```

Comment:
@bluescarni how would you handle such a case (in which the `i` is used for both accessing the vector and also in assigning its element), if not with a cast?

Reply:
@Sceki sorry I was not clear. The cast to double here is ok, as you want to convert that ``i`` to a floating-point anyway as it makes sense for the computation. In this case, I believe GCC is being a bit too strict, as it is complaining that the implicit cast to double might not preserve the exact value of the original variable (you are converting a 64bit integer to double precision, so if the number has a magnitude larger than 2**53 the conversion will not be exact - that's why GCC wants the static cast, to make extra sure that you are ok with the potential loss of precision). 

So keep the cast here, even if it looks ugly.